### DISCUSSION TITLE
Syslog priority being added?

### DISCUSSION THREAD

**Discussion by corycandia** [Security-Onion-Solutions/securityonion#5395] [Unsupported Versions] [answered]
Community,

I can't figure out how to remove priority from getting added to the beginning of message.

Example of logs and the different formats:

<142>2021:09:03-07:45:10 gateway ...
<142>1 2021:09:03-07:45:10 gateway ...
<142> 2021:09:03-07:45:10 gateway ...

It's making it extremely difficult to GROK.

I can't figure out how it's getting added.  The syslog messages that go to VMware Log Insight don't have the priority at the beginning.  They all start with timestamp.  

I can't tell if it's being sent by the Sophos, being added by filebeats, logstash, or elastic search.  I'm searching through salt files looking between defaults and local to see where that might come from?

Any guidance?

**Comment by weslambert** [comment, marked as answer]
The priority should be sent as part of the the original syslog message header by whatever is sending it.

**Comment by abesinger** [comment]
Some more detail...

The priority isn't being "added" by anything, it's included by the originating host and is a required part of the syslog protocol, as defined in [RFC 5254](https://datatracker.ietf.org/doc/html/rfc5424#section-6).  *nix log servers retain the priority when forwarding logs, but most strip the priority when writing logs to local files, at least by default. Elastic of course takes the message as it's sent.


Why do you say it's difficult to GROK?  I use the following GROK pattern to match the priority and timestamp:  `^<%{INT:syslog.priority:int}>%{TIMESTAMP_ISO8601:syslog.timestamp}`.  And if you want to see how to extract the facility and level from the priority, see #5251 

I do notice, however, in the example messages you gave, the 2nd line seems to have a stray "1" after the priority, and before the timestamp:
```
<142>1 2021:09:03-07:45:10 gateway ...
```
That `1` is not part of the priority.  Is it a typo or paste error?  If not and you're getting that in some of your logs, I can see how it might be more difficult to match with GROK, but it should still be do-able.

**Reply by corycandia** [reply]
It was not an accident, that was exactly why I was trying to make sure I understood the original host was sending it.  I'm just using dissection to try and trim stuff off.

**Reply by abesinger** [reply]
Yeah, that's weird.  What's generating the log message?

**Reply by corycandia** [reply]
It originated from a Sophos UTM running 9.6 being forwarded from VMware Log Insight.  The original thought behind forwarding was to try and avoid re-configuring every device in the enterprise.  Just could not figure out how to get that priority to be consistent to trim/dissect/grok, so had to have Sophos direct send.  At least Cisco lets you customize the timestamp components...
